src/display/annotation_layer.js

```diff
@@ -530,6 +539,107 @@ var TextWidgetAnnotationElement = (
 
 /**
  * @class
+ * @alias CheckboxWidgetAnnotationElement
+ */
+var CheckboxWidgetAnnotationElement =
+  (function CheckboxWidgetAnnotationElementClosure() {
+
+  function CheckboxWidgetAnnotationElement(parameters) {
+    WidgetAnnotationElement.call(this, parameters);
+  }
+
+  Util.inherit(CheckboxWidgetAnnotationElement, WidgetAnnotationElement, {
+    /**
+     * Render the text widget annotation's HTML element in the empty container.
+     *
+     * @public
+     * @memberof CheckboxWidgetAnnotationElement
+     * @returns {HTMLSectionElement}
+     */
+    render: function CheckboxWidgetAnnotationElement_render() {
+      this.container.className = 'checkboxWidgetAnnotation';
+
+      if (this.renderInteractiveForms) {
+        var element = document.createElement('div');
+        element.addEventListener('click', this._toggle.bind(this, element));
+        this.container.appendChild(element);
+      }
+
+      return this.container;
+    },
+
+    /**
+     * Toggle the checkbox.
+     *
+     * @private
+     * @memberof CheckboxWidgetAnnotationElement
+     */
+    _toggle: function CheckboxWidgetAnnotationElement_toggle(element) {
+      element.className = element.className ? '' : 'checked';
```

Comment:
Why are we toggling the class name here? I would expect that we set the `checked` *attribute* and style that with CSS (using the `input:checked` selector) so we don't have to use classes here.